src/Features/Core/Portable/PullMemberUp/CodeActionAndSolutionGenerator.cs

```diff
@@ -0,0 +1,197 @@
+            var symbolsToPullUp = result._membersAnalysisResults.
+                Select(analysisResult =>
+                {
+                    if (analysisResult._member is IPropertySymbol propertySymbol)
+                    {
+                        // When it is a property, it could have a public getter/setter
+                        // but other one is not public. In this scenario, only the public getter/setter
+                        // will be add the target interface
+                        return CodeGenerationSymbolFactory.CreatePropertySymbol(
+                                propertySymbol,
+                                getMethod: FilterGetterOrSetter(propertySymbol.GetMethod),
+                                setMethod: FilterGetterOrSetter(propertySymbol.SetMethod));
+                    }
+                    else
+                    {
+                        return analysisResult._member;
+                    }
+                });
+
+            return new DocumentChangeAction(
+                title,
+                cancellationToken => codeGenerationService.AddMembersAsync(
+                    contextDocument.Project.Solution, result._target, symbolsToPullUp, options: options, cancellationToken: cancellationToken));
+        }
+
+        private IMethodSymbol FilterGetterOrSetter(IMethodSymbol getterOrSetter)
+        {
+            return getterOrSetter?.DeclaredAccessibility == Accessibility.Public ? getterOrSetter : null;
+        }
+
+        private Solution PullMembersIntoClass(
+            AnalysisResult result,
+            ICodeGenerationService codeGenerationService,
+            SyntaxNode targetNodeSyntax,
+            SolutionEditor solutionEditor,
+            Solution solution,
+            ImmutableDictionary<DocumentId, DocumentEditor> editorMap,
+            ImmutableDictionary<ISymbol, IEnumerable<SyntaxNode>> syntaxMap)
+        {
+            AddMembersToClass(result, codeGenerationService, targetNodeSyntax, solution, editorMap);
+            RemoveOriginalMembers(result, solution, editorMap, syntaxMap);
+            return solutionEditor.GetChangedSolution();
+        }
+
+        private void AddMembersToClass(
+            AnalysisResult result,
+            ICodeGenerationService codeGenerationService,
+            SyntaxNode targetNodeSyntax,
+            Solution solution,
+            ImmutableDictionary<DocumentId, DocumentEditor> editorMap)
+        {
+            var pullUpMembersSymbols = result._membersAnalysisResults.Select(memberResult => memberResult._member);
+
+            var options = new CodeGenerationOptions(reuseSyntax: true, generateMethodBodies: false);
+            var membersAddedNode = codeGenerationService.AddMembers(targetNodeSyntax, pullUpMembersSymbols, options: options);
+
+            var editor = editorMap[solution.GetDocumentId(targetNodeSyntax.SyntaxTree)];
+            editor.ReplaceNode(targetNodeSyntax, membersAddedNode);
```

Comment:
because of all the edits you are doing to all the files, you lkely need to use the callback for of editor.ReplaceNode, that way this will see the results of any edits you make to the target (and vice-versa).  Note: this is only an issue if there might possibly be other edits lexically within in your targetNodeSyntax.  If not, you don't have to do this.